go/types/struct.go

```diff
@@ -66,6 +66,24 @@ func NewStruct(name string, data StructData) Struct {
 	return validateStruct(newStruct(name, fieldNames, values))
 }
 
+type StructTemplate struct {
+	name       string
+	fieldNames []string
+}
+
+func MakeStructTemplate(name string, fieldNames []string) StructTemplate {
+	st := StructTemplate{name, fieldNames}
+	for i := 0; i < len(st.fieldNames); i++ {
+		verifyFieldName(st.fieldNames[i])
```

Comment:
Also verify the order. There is code for all of this in there. Maybe you need to refactor it a bit?